Add inference section to task guides

stevhliu opened the pull request:
Currently, the task guides only show how to finetune a model, but it doesn't directly connect the dots to how you can use that model for inference. For completeness, this PR adds a section to the task guides to show how to use a model for inference after finetuning. This gives users a better overview of the model lifecycle.

In doing so, when we update `task_summary.mdx`, we can focus less on the practical steps of how to use a model for inference (we can add links to the task guides) and instead discuss the more theoretical aspects of these tasks as intended by the Conceptual Guide section.

stevhliu committed:
📝 start adding inference section to task guides

stevhliu committed:
✨ make style

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_18781). All of your documentation changes will be reflected on that endpoint.

stevhliu committed:
📝 add multiple choice

stevhliu committed:
add rest of inference sections

stevhliu committed:
make style

sgugger commented on docs/source/en/tasks/language_modeling.mdx:
```diff
@@ -303,6 +303,71 @@ Call [`fit`](https://keras.io/api/models/model_training_apis/#fit-method) to fin
+>>> from transformers import AutoTokenizer
+
+>>> tokenizer = AutoTokenizer.from_pretrained("distilgpt2")
+>>> inputs = tokenizer(prompt, return_tensors="pt").input_ids
```
```suggestion
>>> inputs = tokenizer(prompt, return_tensors="tf").input_ids
```

sgugger commented on docs/source/en/tasks/asr.mdx:
````diff
@@ -223,4 +223,52 @@ At this point, only three steps remain:
+```
+
+<frameworkcontent>
+<pt>
````
Why not use the pipeline function, at least at first? Then dive into processing + model.

sgugger commented on docs/source/en/tasks/asr.mdx:
````diff
@@ -223,4 +223,52 @@ At this point, only three steps remain:
+```py
+>>> from transformers import AutoModelForCTC
+
+>>> model = AutoModelForCTC.from_pretrained("facebook/wav2vec2-base-960h")
````
This needs to be the fine-tuned model otherwise it defeats the point of this section.

sgugger commented on docs/source/en/tasks/language_modeling.mdx:
````diff
@@ -303,6 +303,71 @@ Call [`fit`](https://keras.io/api/models/model_training_apis/#fit-method) to fin
+Use the [`~transformers.generation_tf_utils.TFGenerationMixin`] method to generate text. Check out the [Generation](./main_classes/text_generation) to learn more about the different text generation strategies, and parameters for controlling generation.
+
+```py
+>>> from transformers import AutoModelForCausalLM
````
```suggestion
>>> from transformers import TFAutoModelForCausalLM
```

sgugger reviewed (commented):
Thanks for this PR, but I'm not sure the inference section is really useful if it doesn't use the model that was finetuned, as people can find how to use a given pretrained model in the Hub directly.

Also, it would be great to show how to use the fine-tuned model in a pipeline first, which is end-to-end, then dive into the details. I left comments on the first task, but this applies to all of them.

Finally, the TensorFlow content is not TensorFlow: it still uses the same code as PyTorch.

stevhliu committed:
add compute_metric, push_to_hub, pipeline

stevhliu committed:
make style

stevhliu commented:
Thanks for the suggestions! I reworked the `sequence_classification` task a bit, and if we like the changes, then I can apply them to the other tasks. Main changes are:

- Encourage users to login with their HF accounts so they can push their finetuned models. Along these lines, I've added `push_to_hub` and the `PushToHubCallback`.
- Added a section to include a `compute_metrics` function so users can evaluate their models during training.

I think adding these two will help the task guides be more complete :)

sgugger commented on docs/source/en/tasks/sequence_classification.mdx:
````diff
@@ -197,18 +251,163 @@ Configure the model for training with [`compile`](https://keras.io/api/models/mo
+>>> text = "This was a masterpiece. Not completely faithful to the books, but enthralling from beginning to end. Might be my favorite of the three."
+```
+
+The simplest way to try out your finetuned model for inference is to use it in a [`pipeline`]. Map the expected id to their labels with `id2label` and `label2id`:
+
+```py
+>>> id2label = {0: "NEGATIVE", 1: "POSITIVE"}
+>>> label2id = {"NEGATIVE": 0, "POSITIVE": 1}
+```
````
This should be done before training saving/pushing to the Hub ideally. So the saved/uploaded model is correct.

sgugger commented on docs/source/en/tasks/sequence_classification.mdx:
````diff
@@ -197,18 +251,163 @@ Configure the model for training with [`compile`](https://keras.io/api/models/mo
+Instantiate the `pipeline` for sentiment analysis with your model and tokenizer, and pass your text to it:
+
+```py
+>>> classifier = pipeline("sentiment-analysis", model=model, tokenizer=tokenizer)
````
```suggestion
>>> classifier = pipeline("sentiment-analysis", model="stevhliu/my_awesome_model")
```
If labels are handled previously, you'll be able to do just this, which is way shorted (and framewok-agnostic).

sgugger reviewed (commented):
Thanks! I think this is way better this way. Just have a couple of comments on the way labels are handled but otherwise changes are good to push to the other tasks!

stevhliu committed:
add updated sequence and token classification

stevhliu committed:
make style

stevhliu committed:
make edits in token classification

stevhliu committed:
add audio classification

stevhliu committed:
make style

stevhliu committed:
add asr

stevhliu committed:
make style

stevhliu committed:
add image classification

stevhliu committed:
make style

stevhliu committed:
add summarization

stevhliu committed:
make style

stevhliu committed:
add translation

stevhliu committed:
make style

stevhliu committed:
add multiple choice

stevhliu committed:
add language modeling

stevhliu committed:
add qa

stevhliu committed:
make style

stevhliu committed:
review and edits

sgugger commented on docs/source/en/tasks/translation.mdx:
```diff
@@ -14,30 +14,47 @@ specific language governing permissions and limitations under the License.
 <Youtube id="1JvfrvZgi6c"/>
 
-Translation converts a sequence of text from one language to another. It is one of several tasks you can formulate as a sequence-to-sequence problem, a powerful framework that extends to vision and audio tasks. 
+Translation converts a sequence of text from one language to another. It is one of several tasks you can formulate as a sequence-to-sequence problem, a powerful framework for returning some output from an input like translation or summarization. Companies like Google use sequence-to-sequence models to power its products like Translate and Docs.
```
Why are we doing free advertisement for Google here?

sgugger commented on docs/source/en/tasks/translation.mdx:
````diff
@@ -82,84 +99,113 @@ The preprocessing function needs to:
 >>> tokenized_books = books.map(preprocess_function, batched=True)
 ```
 
+Now create a batch of examples using [`DataCollatorForSeq2Seq`]. It will also *dynamically pad* your text and labels to the length of the longest element in its batch (instead of the entire dataset) so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by setting `padding=True`, dynamic padding is more efficient.
````
The second sentence is not accurate. It's more efficient to dynamically pad the sentences to the longest length in the batch during collation, rather than padding the whole dataset to the maximum length once.

sgugger commented on docs/source/en/tasks/translation.mdx:
````diff
@@ -169,13 +215,41 @@ At this point, only three steps remain:
+To finetune a model in TensorFlow, start by setting up an optimizer function, learning rate schedule, and some training hyperparameters:
+
+```py
+>>> from transformers import create_optimizer, AdamWeightDecay
````
It doesn't look like the `create_optimizer` function is used.

sgugger commented on docs/source/en/tasks/token_classification.mdx:
````diff
@@ -126,13 +145,13 @@ Here is how you can create a function to realign the tokens and labels, and trun
 ```
 
-Use [`DataCollatorForTokenClassification`] to create a batch of examples. It will also *dynamically pad* your text and labels to the length of the longest element in its batch, so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by setting `padding=True`, dynamic padding is more efficient.
+Now create a batch of examples using [`DataCollatorWithPadding`]. It'll also *dynamically pad* your text and labels to the length of the longest element in its batch (instead of the entire dataset) so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by setting `padding=True`, dynamic padding is more efficient.
````
AS mentioned in the translation guide, this is a bit misleading. It's more efficient to pad dynamically to the max length in the batch vs padding to the max length of the dataset, but in both cases this uses the `tokenizer` pad method behind the scenes.

sgugger commented on docs/source/en/tasks/token_classification.mdx:
````diff
@@ -151,39 +170,120 @@ Use [`DataCollatorForTokenClassification`] to create a batch of examples. It wil
+If you aren't familiar with finetuning a model with the [`Trainer`], take a look at the basic tutorial [here](../training#train-with-pytorch-trainer)!
 
 </Tip>
+You're ready to start training your model now! Map the expected ids to their labels with `id2label` and `label2id`:
+
+```py
+>>> id2label = {
+...     0: "O",
+...     1: "B-corporation",
+...     2: "I-corporation",
+...     3: "B-creative-work",
+...     4: "I-creative-work",
+...     5: "B-group",
+...     6: "I-group",
+...     7: "B-location",
+...     8: "I-location",
+...     9: "B-person",
+...     10: "I-person",
+...     11: "B-product",
+...     12: "I-product",
+... }
+>>> label2id = {
+...     "O": 0,
+...     "B-corporation": 1,
+...     "I-corporation": 2,
+...     "B-creative-work": 3,
+...     "I-creative-work": 4,
+...     "B-group": 5,
+...     "I-group": 6,
+...     "B-location": 7,
+...     "I-location": 8,
+...     "B-person": 9,
+...     "I-person": 10,
+...     "B-product": 11,
+...     "I-product": 12,
+... }
+```
````
This part is framework-agnostic so could go before the split PT/TF.

sgugger commented on docs/source/en/tasks/token_classification.mdx:
````diff
@@ -239,12 +327,67 @@ Set up an optimizer function, learning rate schedule, and some training hyperpar
+Map the expected ids to their labels with `id2label` and `label2id`:
+
+```py
+>>> id2label = {
````
Same here, this would avoid this dupe code.

sgugger reviewed (commented):
Reviewed the last two guides for now, will continue with the other ones later.

sgugger commented on docs/source/en/tasks/language_modeling.mdx:
````diff
@@ -156,11 +171,11 @@ Apply the `group_texts` function over the entire dataset:
 ```
 
-For causal language modeling, use [`DataCollatorForLanguageModeling`] to create a batch of examples. It will also *dynamically pad* your text to the length of the longest element in its batch, so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by setting `padding=True`, dynamic padding is more efficient. 
+Now create a batch of examples using [`DataCollatorForLanguageModeling`]. It'll also *dynamically pad* your text to the length of the longest element in its batch (instead of the entire dataset) so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by setting `padding=True`, dynamic padding is more efficient. 
````
Same comment as before on the tokenizer here.

sgugger commented on docs/source/en/tasks/multiple_choice.mdx:
````diff
@@ -79,15 +100,15 @@ The preprocessing function needs to do:
 ```
 
-🤗 Transformers doesn't have a data collator for multiple choice, so you will need to create one. You can adapt the [`DataCollatorWithPadding`] to create a batch of examples for multiple choice. It will also *dynamically pad* your text and labels to the length of the longest element in its batch, so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by set ...
+🤗 Transformers doesn't have a data collator for multiple choice, so you'll need to adapt the [`DataCollatorWithPadding`] to create a batch of examples. It'll also *dynamically pad* your text and labels to the length of the longest element in its batch (instead of the entire dataset) so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by setting `padding ...
````
Same here

sgugger commented on docs/source/en/tasks/sequence_classification.mdx:
````diff
@@ -91,38 +110,74 @@ Use [`DataCollatorWithPadding`] to create a batch of examples. It will also *dyn
+If you aren't familiar with finetuning a model with the [`Trainer`], take a look at the basic tutorial [here](../training#train-with-pytorch-trainer)!
 
 </Tip>
+You're ready to start training your model now! Map the expected ids to their labels with `id2label` and `label2id`:
+
+```py
+>>> id2label = {0: "NEGATIVE", 1: "POSITIVE"}
+>>> label2id = {"NEGATIVE": 0, "POSITIVE": 1}
````
This part should go before we split between frameworks.

sgugger commented on docs/source/en/tasks/sequence_classification.mdx:
````diff
@@ -66,13 +85,13 @@ Create a preprocessing function to tokenize `text` and truncate sequences to be
 ```
 
-Use [`DataCollatorWithPadding`] to create a batch of examples. It will also *dynamically pad* your text to the length of the longest element in its batch, so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by setting `padding=True`, dynamic padding is more efficient.
+Now create a batch of examples using [`DataCollatorWithPadding`]. It'll *dynamically pad* your text to the length of the longest element in its batch (instead of the entire dataset) so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by setting `padding=True`, dynamic padding is more efficient.
````
Same comment as before here.

sgugger commented on docs/source/en/tasks/summarization.mdx:
````diff
@@ -84,13 +104,13 @@ The preprocessing function needs to:
 ```
 
-Use [`DataCollatorForSeq2Seq`] to create a batch of examples. It will also *dynamically pad* your text and labels to the length of the longest element in its batch, so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by setting `padding=True`, dynamic padding is more efficient.
+Now create a batch of examples using [`DataCollatorForSeq2Seq`]. It will also *dynamically pad* your text and labels to the length of the longest element in its batch (instead of the entire dataset) so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by setting `padding=True`, dynamic padding is more efficient.
````
Same comment here.

sgugger reviewed (commented):
Finished reviewing the other tasks. Thanks for putting this all together!

stevhliu commented on docs/source/en/tasks/question_answering.mdx:
```diff
@@ -254,18 +282,349 @@ Configure the model for training with [`compile`](https://keras.io/api/models/mo
+
+You can also manually replicate the results of the `pipeline` if you'd like:
+
+<frameworkcontent>
```
I didn't think this section was too long, and it'd be really cool for users to see how they can replicate the pipeline in another way if they wanted to with their own model. I'll keep this here and remove the evaluation section :)

stevhliu commented on docs/source/en/tasks/asr.mdx:
```diff
@@ -99,32 +119,32 @@ The [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) dataset has a sam
 ...     audio = batch["audio"]
 
-...     batch = processor(audio=audio["array"], sampling_rate=audio["sampling_rate"]).input_values[0]
+...     batch["input_values"] = processor(audio["array"], sampling_rate=audio["sampling_rate"]).input_values[0]
 ...     batch["input_length"] = len(batch["input_values"])
 
 ...     batch["labels"] = processor(text=batch["transcription"]).input_ids
```
Super cool! I tried passing it in one call, but I get a `KeyError: 'labels'` (I included a [notebook](https://colab.research.google.com/drive/1Sa0VU9GeABUieYjX2B9SUoK_Fh5JlkxS?usp=sharing)).

sgugger replied:
The first line needs to be
```
batch = processor(audio["array"], sampling_rate=audio["sampling_rate"], text=batch["transcription"])
```
but then it worked for me.

stevhliu replied:
Ok, this should be fixed now thanks to @sanchit-gandhi ! 👏

stevhliu committed:
apply reviews

stevhliu committed:
make style

stevhliu committed:
fix call to processor

sanchit-gandhi commented on docs/source/en/tasks/asr.mdx:
```diff
@@ -14,27 +14,44 @@ specific language governing permissions and limitations under the License.
 <Youtube id="TksaY_FDgnk"/>
 
-Automatic speech recognition (ASR) converts a speech signal to text. It is an example of a sequence-to-sequence task, going from a sequence of audio inputs to textual outputs. Voice assistants like Siri and Alexa utilize ASR models to assist users.
+Automatic speech recognition (ASR) converts a speech signal to text, going from a sequence of audio inputs to textual outputs. Virtual assistants like Siri and Alexa use ASR models to help users everyday, and there are many other useful user-facing applications like live captioning and note-taking during meetings.
```
`textual` sounds a bit funky to me!

```suggestion
Automatic speech recognition (ASR) converts a speech signal to text, mapping a sequence of audio inputs to text outputs. Virtual assistants like Siri and Alexa use ASR models to help users everyday, and there are many other useful user-facing applications like live captioning and note-taking during meetings.
```

sanchit-gandhi commented on docs/source/en/tasks/asr.mdx:
```diff
@@ -14,27 +14,44 @@ specific language governing permissions and limitations under the License.
-This guide will show you how to fine-tune [Wav2Vec2](https://huggingface.co/facebook/wav2vec2-base) on the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) dataset to transcribe audio to text.
+This guide will show you how to:
+
+1. Finetune [Wav2Vec2](https://huggingface.co/facebook/wav2vec2-base) on the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) to transcribe audio to text.
```
```suggestion
1. Finetune [Wav2Vec2](https://huggingface.co/facebook/wav2vec2-base) on the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) dataset to transcribe audio to text.
```

sanchit-gandhi commented on docs/source/en/tasks/asr.mdx:
````diff
@@ -14,27 +14,44 @@ specific language governing permissions and limitations under the License.
+Before you begin, make sure you have all the necessary libraries installed:
+
+```bash
+pip install transformers datasets evaluate jiwer
````
Just checking that this example works without an audio processing library like `soundfile` or `librosa`? Great if it does! Some of these libraries have an annoying installation on Linux

stevhliu replied:
Yeah I was able to run the code without any issues :)

sanchit-gandhi commented on docs/source/en/tasks/asr.mdx:
```diff
@@ -14,27 +14,44 @@ specific language governing permissions and limitations under the License.
 ## Load MInDS-14 dataset
 
-Load the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) from the 🤗 Datasets library:
+Start by loading a smaller subset of the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) dataset from the 🤗 Datasets library. This'll give you a chance to experiment and make sure everythings works before spending more time training on the full dataset.
```
```suggestion
Start by loading a smaller subset of the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) dataset from the 🤗 Datasets library. This'll give you a chance to experiment and make sure everything works before spending more time training on the full dataset.
```

sanchit-gandhi commented on docs/source/en/tasks/asr.mdx:
```diff
@@ -99,32 +119,28 @@ The [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) dataset has a sam
-...     batch["input_length"] = len(batch["input_values"])
-
-...     batch["labels"] = processor(text=batch["transcription"]).input_ids
+...     batch = processor(audio["array"], sampling_rate=audio["sampling_rate"], text=batch["transcription"])
```
Nice!

sanchit-gandhi commented on docs/source/en/tasks/asr.mdx:
```diff
@@ -212,15 +264,89 @@ At this point, only three steps remain:
+
+>>> transcriber = pipeline("automatic-speech-recognition", model="stevhliu/my_awesome_asr_minds_model")
+>>> transcriber(audio_file)
+{'text': "ALDCBC</s>COCB</s>DCBDBCBRBCDCTCRWKDBK<s></s>KODT</s>C</s>T</s>C</s>OD<s>B</s>BCTOR</s>AKDOBODTCTEDQOATDVDKCDSKL<unk>KEKEPKNPKPADSRDSCLCQ<unk>ED<s>BDXAXDP<unk>KQDPWDKLEPRPKDTDRDPDNFRTADPLDSRDPDKDX<unk>XLXCXLCL</s>LCXLCLDT</s>TBLDLBLTLCB</s>BXTLTBLDTM</s>BLBLCLBLCLDT</s>LXCDTB</s>L</s>T</s>LTDK</s>TDTDTLOKDADPDSDT</s><unk>RDQTDDADSNDN</s>KD</s>T'DPNVPHKDPDRDRKRKD</s>PKDCQS<unk>X<unk>LTAQ ...
```
Maybe we can train on a few hundred samples to get a somewhat reasonable transcription? Or is the point of this .md purely to demonstrate **how** to fine-tune (rather than with any care to performance), meaning we're happy with mumbo jumbo here?

stevhliu replied:
I think I may have run out of memory in Colab when I tried with more samples, but I'll try again with a few hundred samples to see if we can do better than this gibberish.

sanchit-gandhi replied:
Note that when you download a dataset using `train[:20]` you actually download the entire dataset and then slice the first 20 samples -> so in terms of disk space using a couple of hundred samples will be equivalent to just 20.

For training, you might be getting OOM's because there are very long audio samples in your dataset. The self-attention mechanism memory scales with sequence length squared: $\text{memory} \propto \text{seq len}^2$. So there could be long samples > 20s that are causing OOM's! We usually filter such samples by a 20s max length criteria: 
https://github.com/huggingface/transformers/blob/a0f867430347bcf939f71d186409b9ca138c3b34/examples/pytorch/speech-recognition/run_speech_recognition_ctc.py#L634-L642

stevhliu replied:
Finally trained a better model that doesn't return mumbo jumbo and added an important note on making sure the dataset text matches the casing of the tokenizer!

sanchit-gandhi replied:
Nice! Glad to hear it's working 🗣

sanchit-gandhi commented on docs/source/en/tasks/audio_classification.mdx:
```diff
@@ -14,27 +14,44 @@ specific language governing permissions and limitations under the License.
-This guide will show you how to fine-tune [Wav2Vec2](https://huggingface.co/facebook/wav2vec2-base) on the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) to classify intent.
+This guide will show you how to:
+
+1. Finetune [Wav2Vec2](https://huggingface.co/facebook/wav2vec2-base) on the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) to classify speaker intent.
```
```suggestion
1. Finetune [Wav2Vec2](https://huggingface.co/facebook/wav2vec2-base) on the [MInDS-14](https://huggingface.co/datasets/PolyAI/minds14) dataset to classify speaker intent.
```

sanchit-gandhi commented on docs/source/en/tasks/audio_classification.mdx:
```diff
@@ -14,27 +14,44 @@ specific language governing permissions and limitations under the License.
 <Youtube id="KWwzcmG98Ds"/>
 
-Audio classification assigns a label or class to audio data. It is similar to text classification, except an audio input is continuous and must be discretized, whereas text can be split into tokens. Some practical applications of audio classification include identifying intent, speakers, and even animal species by their sounds.
+Audio classification assigns a label or class to audio data. It is similar to text classification, except an audio input is continuous and must be discretized, whereas text can be split into tokens. Some practical applications of audio classification include identifying speaker intent, language classification, and even animal species by their sounds.
```
Actually think that the continuous / discrete statement is a bit confusing? IMO we want to drive the similarity between text and audio classification:
- Text input -> class label output
- Audio input -> class label output

=> same task, different input domain

stevhliu replied:
Good point! 👍

sanchit-gandhi commented on docs/source/en/tasks/audio_classification.mdx:
````diff
@@ -151,25 +199,28 @@ Load Wav2Vec2 with [`AutoModelForAudioClassification`]. Specify the number of la
 ```py
 >>> training_args = TrainingArguments(
-...     output_dir="./results",
+...     output_dir="my_awesome_mind_model",
````
Maybe a different name to the ASR one?
```suggestion
...     output_dir="my_awesome_mind_classification_model",
```

stevhliu replied:
The ASR one is `my_awesome_asr_mind_model` which I think is different enough

sanchit-gandhi commented on docs/source/en/tasks/audio_classification.mdx:
````diff
@@ -178,15 +229,89 @@ At this point, only three steps remain:
+```py
+>>> from datasets import load_dataset, Audio
+
+>>> dataset = load_dataset("narad/ravdess", split="train")
````
Do we want to use this dataset instead of minds for inference?

stevhliu replied:
Oops, completely wrong dataset! Thanks for catching this :)

sanchit-gandhi commented on docs/source/en/tasks/asr.mdx:
````diff
@@ -157,17 +173,55 @@ Unlike other data collators, this specific data collator needs to apply a differ
 ```
 
-Create a batch of examples and dynamically pad them with `DataCollatorForCTCWithPadding`:
+Now create a batch of examples and dynamically pad them with `DataCollatorForCTCWithPadding`:
````
This is potentially a bit misleading since we don't actually batch or pad examples here, but simply instantiate the data collator

sanchit-gandhi reviewed (commented):
Audio sections look good! Just some minor formatting suggestions from me!

stevhliu committed:
apply audio reviews

stevhliu committed:
update to better asr model

stevhliu committed:
make style
